packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/update.ts

```diff
@@ -327,3 +333,15 @@ export const performUpdate = async <T>(
     shouldPerformUpsert ? upsert : attributes
   );
 };
+
+const recursiveMerge = (target: Record<string, any>, value: any, keys: string[] = []) => {
```

Comment:
nice! it still doesn't handle the saved_searches case though.

Reply:
Every FTR test is green, so I assume it does?